pkg/networkmanager/openvpn.jsx

```diff
@@ -0,0 +1,112 @@
+            await cockpit.file(userCertPath).replace(userCertVal);
+            await cockpit.file(userKeyPath).replace(userPrivateKey);
+        } catch (err) {
+            console.log("an error occured => ", err);
+        }
+
+        function createSettingsObject() {
+            return {
+                ...settings,
+                connection: {
+                    ...settings.connection,
+                    type: 'vpn',
+                },
+                vpn: {
+                    data: {
+                        remote,
+                        ca: caPath,
+                        cert: userCertPath,
+                        key: userKeyPath,
+                        // hardcoding the bellow properties until we have a design for advanced dialog
+                        'cert-pass-flags': '0',
+                        'connection-type': 'tls',
+                        dev: 'tun',
+                        'push-peer-info': 'yes',
+                        'remote-cert-tls': 'server'
+                    },
+                    'service-type': 'org.freedesktop.NetworkManager.openvpn'
+                }
+            };
+        }
+
+        dialogSave({
+            connection,
+            dev,
+            model,
+            settings: createSettingsObject(),
+            onClose: Dialogs.close,
+            setDialogError,
+        });
+    }
+
+    return (
+        <NetworkModal
+            title={!connection ? _("Add OpenVPN") : _("Edit OpenVPN settings")}
+            isCreateDialog={!connection}
+            onSubmit={onSubmit}
+            dialogError={dialogError}
+            idPrefix={idPrefix}
+        >
+            <Name idPrefix={idPrefix} iface={iface} setIface={setIface} />
+            <FormGroup label={_("Remote")}>
+                <TextInput id={idPrefix + '-remote-input'} value={remote} onChange={(_, val) => setRemote(val)} />
+            </FormGroup>
+            <FormGroup label={_("CA certificate")}>
+                <FileUpload id={idPrefix + '-ca'} filename={caCertName} onFileInputChange={(_, file) => setCaCertName(file.name)} type='text' onDataChange={(_, val) => setCaCertVal(val)} hideDefaultPreview />
+            </FormGroup>
+            <FormGroup label={_("User certificate")}>
+                <FileUpload id={idPrefix + '-user-cert'} filename={userCertName} onFileInputChange={(_, file) => setUserCertName(file.name)} type='text' onDataChange={(_, val) => setUserCertVal(val)} hideDefaultPreview />
+            </FormGroup>
+            <FormGroup label={_("User private key")}>
+                <FileUpload id={idPrefix + '-user-key'} filename={userKeyName} onFileInputChange={(_, file) => setUserKeyName(file.name)} type='text' onDataChange={(_, val) => setUserPrivateKey(val)} hideDefaultPreview />
+            </FormGroup>
+            <FormGroup label={_("Test")}>
+                <input type='file' id={idPrefix + '-test-upload'} />
+            </FormGroup>
+        </NetworkModal>
```

Comment:
These 17 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19363-20230921-075022-79cb9fd2-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)